lib/src/async_cache.dart

```diff
@@ -43,70 +48,76 @@ class AsyncCache<T> {
   ///
   /// An ephemeral cache guarantees that a callback function will only be
   /// executed at most once concurrently. This is useful for requests which data
-  /// is updated frequently but the cost of making multiple requests without
-  /// previous completing yet is prohibitive or unwanted.
+  /// is updated frequently but stale data is acceptable.
   ///
   /// For _testing_, a custom `now` function may be provided.
   factory AsyncCache.ephemeral({DateTime now()}) =>
       new AsyncCache._(Duration.ZERO);
 
   // Prevent inheritance. This allows using a redirecting factory constructor
   // with different implementations in the future without a breaking API change.
-  AsyncCache._(this._duration, {DateTime now(): _defaultNow,})
+  AsyncCache._(
+    this._duration, {
+    DateTime now(): _defaultNow,
+  })
       : _getNow = now;
 
   /// Returns a cached value or runs [callback] to compute a new one.
   ///
   /// If [callback] has been run recently enough, returns its previous return
   /// value. Otherwise, runs [callback] and returns its new return value.
   Future<T> fetch(Future<T> callback()) async {
+    if (_cachedStreamSplitter != null) {
+      throw new StateError('Previously used to cache via `fetchStream`');
+    }
     _invalidateWhenStale();
-    var result = await (_cachedValueFuture ??= callback());
+    if (_cachedValueFuture == null) {
+      _cachedValueFuture = callback();
+    }
+    var result = await _cachedValueFuture;
     _cachedExpiration ??= _getNow().add(_duration);
```

Comment:
I was thinking this would look like:

```dart
if (_cachedValueFuture == null) {
  _cachedValueFuture = callback();
  await _cachedValueFuture;
  _cachedExpiration = _getNow().add(_duration);
}
```

so it's clear that `_cachedExpiration` is reset if and only if there wasn't a cached value.